components/calendar/generateCalendar.tsx

```diff
@@ -198,10 +201,14 @@ function generateCalendar<DateType>(generateConfig: GenerateConfig<DateType>) {
       triggerPanelChange(mergedValue, newMode);
     };
 
-    const onInternalSelect = (date: DateType) => {
+    const onInternalSelect = (date: DateType, selectType?: CalendarSelectable) => {
       triggerChange(date);
 
-      onSelect?.(date);
+      if (selectable && selectable.length > 0 && selectType) {
+        if (!selectable.includes(selectType)) return;
```

Comment:
```suggestion
        if (!selectable.includes(selectType)) {
          return;
        }
```

Reply:
I had to add the exist check on `selectable` due to linting error